core/services/pipeline/scheduler.go

```diff
@@ -195,6 +196,17 @@ Loop:
 			s.vars.Set(result.Task.DotID(), result.Result.Value)
 		}
 
+		// if the task was marked as fail early, and the result is a fail
+		if result.Result.Error != nil && result.Task.Base().FailEarly == "true" {
+			// drain remaining jobs then exit
```

Comment:
Where are the jobs drained?

Reply:
This loop, it keeps iterating and awaiting all running jobs to complete, but it won't schedule any new jobs.